ocaml/xapi/xapi_vdi.ml

```diff
@@ -128,6 +152,9 @@ let check_operation_error ~__context ?(sr_records=[]) ?(pbd_records=[]) ?(vbd_re
       then Some (Api_errors.vdi_in_use,[_ref; (Record_util.vdi_operation_to_string op)])
       else if my_has_current_operation_vbd_records <> []
       then Some (Api_errors.other_operation_in_progress, [ "VDI"; _ref ])
+      else let sm_feature_error = check_sm_feature_error op sm_features sr in
```

Comment:
Same as the above comment, we could bring the definition of the `sm_features` variable here, as it's not used before this point, and it's only used in this expression:
```ocaml
else
let sm_features = Xapi_sr_operations.features_of_sr_internal ~__context ~_type:sr_type in
let sm_feature_error = check_sm_feature_error op sm_features sr in
if sm_feature_error <> None
then sm_feature_error
else
```